### PR TITLE
Wait for the initialized language worker to be added to webhost worker channels collection

### LINKED ISSUE #4559: Functions fail with "Number of initialized language workers exceeds:2"
For dedicated Java function apps, we see some functions failing with the following error:

> ErrorSystem.InvalidOperationException : Number of initialized language workers exceeded:2 exceeded maxProcessCount: 1   

This is due to a race condition that starts 2 java.exe processes when we only expect one.

Workaround:
Since it is a race condition, sometimes restarting the app gets it in a healthy state.

If this does not work, you can try removing `FUNCTIONS_WORKER_RUNTIME` until the fix is rolled out

cc: @pragnagopa

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #4559

### CODE REVIEW

**Comment by pragnagopa** [comment]
Synced up with @brettsam offline.  I can repro this issue locally.  Verified issue does not repro with this fix.

**Comment by fabiocav** [comment]
Can you please rebase this changes?

**Comment by pragnagopa** [comment]
Rebased. Please review.

File: src/WebJobs.Script/Rpc/LanguageWorkerChannelManager.cs [hunk 1]
```diff
@@ -48,8 +47,6 @@ public class LanguageWorkerChannelManager : ILanguageWorkerChannelManager
 
             _shutdownStandbyWorkerChannels = ScheduleShutdownStandbyChannels;
             _shutdownStandbyWorkerChannels = _shutdownStandbyWorkerChannels.Debounce(5000);
-            _rpcChannelReadySubscriptions = _eventManager.OfType<RpcWebHostChannelReadyEvent>()
-               .Subscribe(AddOrUpdateWorkerChannels);
```
Comment (brettsam) [inline src/WebJobs.Script/Rpc/LanguageWorkerChannelManager.cs]: removing this subscription won't break anything?

Comment (pragnagopa) [reply src/WebJobs.Script/Rpc/LanguageWorkerChannelManager.cs]: Nope. This subscription was specifically being used to add workerchannel to the cache.

**Comment by pragnagopa** [comment]
rebased!

**Pull request merged**